codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -318,6 +321,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
             // If this connection is closing and there are no longer any
             // active streams, close after the current operation completes.
             if (closeListener != null && connection().numActiveStreams() == 0) {
+                ChannelFutureListener closeListener = Http2ConnectionHandler.this.closeListener;
+                // This method could be reentrant, and we don't want to notify the closeListener multiple times
+                Http2ConnectionHandler.this.closeListener = null;
```

Comment:
> What I'm really trying to protect against is multiple calls

Let's just say that in the comment then? I think nulling the `closeListener` makes perfect sense, but the "reentrant" confused me.

> but reentrance is just a superset of that.

subset :stuck_out_tongue_winking_eye: